### ISSUE TITLE
Godot Editor crashes build/run when .csproj is properly placed in subdir

### ISSUE THREAD

**Issue by FyiurAmron** [godotengine/godot#42741]
**Godot version:**
3.2.3-stable_mono_win64

**OS/device including version:**
Windows 7

**Issue description:**
Trying to use `.csproj` placed in a subdir (with proper `.sln` path provided) causes a following error in Godot Editor (e.g. Rider works OK)
```
modules/mono/mono_gd/gd_mono_utils.cpp:371 - Microsoft.Build.Exceptions.InvalidProjectFileException: The project file could not be loaded. Could not find file "C:\xxx\projectName\projectName.csproj"  C:\xxx\projectName\projectName.csproj
 editor/editor_node.cpp:5408 - An EditorPlugin build callback failed.
```
Worth to note is that this setup (`.csproj` in a subdir) is, AFAIK, completely valid and normal for C# projects, even if not used that often in the wild. Also, the path shown in the error does not correspond to path that should be generated based on `.sln` data.

**Steps to reproduce:**
1. create a new C#-ish Godot project
2. move the `.csproj` to any subdir, updating `.sln` to point at it properly
3. test under e.g. Rider (works OK - compiles/builds, runs etc.)
4. test under Godot (crashes with the above error message)

**Minimal reproduction project:**
[CsprojNotFound.zip](https://github.com/godotengine/godot/files/5366039/CsprojNotFound.zip)

**Comment by Calinou** [comment]
> Worth to note is that this setup (.csproj in a subdir) is, AFAIK, completely valid and normal for C# projects, even if not used that often in the wild. Also, the path shown in the error does not correspond to path that should be generated based on .sln data.

This workflow isn't supported in Godot yet. See https://github.com/godotengine/godot-proposals/issues/1575.

**Comment by FyiurAmron** [comment]
Since this feature was included both in 3.5.x, 3.6.x & 4.0.x (as it's a part of https://github.com/godotengine/godot/pull/64089), I'm closing this one.
